frontend/src/components/RestaurantSubMenuButton.tsx

```diff
@@ -0,0 +1,28 @@
+"use client";
+import React from 'react'
+import {useRef} from 'react';
+interface RestaurantSubMenuButtonProps{
+    text: string;
+    setCurrentSelection: (selection: string) => void;
+    selected: boolean;
+}
+export default function RestaurantSubMenuButton({text, setCurrentSelection, selected}: RestaurantSubMenuButtonProps) {
+    const buttonRef = useRef<HTMLButtonElement>(null);
```

Comment:
Hmmm we've done this before. Can we abstract it into a common component so we don't have to keep reusing the same code?

Reply:
Done. This has been extracted away and implemented in all components needing selectability, and removed all unused files now.